File: src/model_v4/entities/SBS_SEGMENT.ts
import { Column, Entity, Index, JoinColumn, ManyToOne, OneToMany, PrimaryColumn } from "typeorm";
import { SBS_AP } from "./SBS_AP";
import { SBS_FILIALE } from "./SBS_FILIALE";


@Entity("SBS_SEGMENT",{schema:"sbsdb"})
@Index("FK8044EDB8427CC73D",["fILIALE_INDEX",])
export class SBS_SEGMENT {

    @PrimaryColumn("bigint",{ 
        nullable:false,
        primary:true,
        name:"SEGMENT_INDEX"
        })
    SEGMENT_INDEX:string;
        

    @Column("bigint",{ 
        nullable:false,
        name:"NETMASK"
        })
    NETMASK:string;
        

    @Column("varchar",{ 
        nullable:false,
        length:50,
        name:"SEGMENT_NAME"
        })
    SEGMENT_NAME:string;
        

    @Column("bigint",{ 
        nullable:false,
        name:"TCP"
        })
    TCP:string;
        

   
    @ManyToOne(type=>SBS_FILIALE, SBS_FILIALE=>SBS_FILIALE.sBS_SEGMENTs,{ onDelete: 'RESTRICT',onUpdate: 'RESTRICT' })
    @JoinColumn({ name:'FILIALE_INDEX'})
    fILIALE_INDEX:SBS_FILIALE | null;


   
    @OneToMany(type=>SBS_AP, SBS_AP=>SBS_AP.sEGMENT_INDEX,{ onDelete: 'RESTRICT' ,onUpdate: 'RESTRICT' })
    sBS_APs:SBS_AP[];
    
}

File: SbsdbServer/sbsdbv4/model/SbsSegment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace hb.SbsdbServer.sbsdbv4.model
{
    [Table("SBS_SEGMENT")]
    public partial class SbsSegment
    {
        public SbsSegment()
        {
            SbsAp = new HashSet<SbsAp>();
        }

        [Column("SEGMENT_INDEX", TypeName = "bigint(20)")]
        public long SegmentIndex { get; set; }
        [Column("NETMASK", TypeName = "bigint(20)")]
        public long Netmask { get; set; }
        [Required]
        [Column("SEGMENT_NAME", TypeName = "varchar(50)")]
        public string SegmentName { get; set; }
        [Column("TCP", TypeName = "bigint(20)")]
        public long Tcp { get; set; }
        [Column("FILIALE_INDEX", TypeName = "bigint(20)")]
        public long? FilialeIndex { get; set; }

        [ForeignKey("FilialeIndex")]
        [InverseProperty("SbsSegment")]
        public virtual SbsFiliale FilialeIndexNavigation { get; set; }
        [InverseProperty("SegmentIndexNavigation")]
        public virtual ICollection<SbsAp> SbsAp { get; set; }
    }
}
